sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,217 @@ class TableCheckpointStore:
+    def from_connection_string(cls, conn_str, table_name, credential=None, **kwargs):
+        """Create TableCheckpointStore from a storage connection string.
+        :param str conn_str:
+            A connection string to an Azure Storage account.
+        :param table_name:
+            The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+        :keyword str api_version:
+            The Storage API version to use for requests. Default value is '2019-07-07'.
+        :keyword str secondary_hostname:
+        :returns: A table checkpoint store.
+        :rtype: ~azure.eventhub.extensions.checkpointstoretable.TableCheckpointStore
+        """
+        endpoint, credential = parse_connection_str(
+            conn_str=conn_str, credential=None, keyword_args=kwargs
+        )
+        return cls(endpoint, table_name=table_name, credential=credential, **kwargs)
+
+    def _checkpoint_entity(self, checkpoint):
+        my_new_entity = {
+            u'PartitionKey': "{} {} {} Checkpoint".format(checkpoint["fullyqualifiednamespace"],
+        checkpoint["eventhubname"], checkpoint['consumergroup']),
+            u'RowKey': checkpoint['partitionid'],
+            u'offset' : checkpoint['offset'],
+            u'sequencenumber' : checkpoint['sequencenumber'],
+        }
+        self.table_client.create_entity(entity=my_new_entity)
+
+    def _ownership_entity(self, ownership):
+        my_new_entity = {
+            u'PartitionKey': "{} {} {} Ownership".format(ownership["fullyqualifiednamespace"],
+        ownership["eventhubname"], ownership['consumergroup']),
+            u'RowKey': ownership['partitionid'],
+            u'ownerid' : ownership['ownerid'],
+        }
+        return self.table_client.create_entity(entity=my_new_entity)
+
+    @classmethod
+    def _create_ownership_entity(cls, ownership):
+        """
+        create a dictionary with the new ownership attributes so that it can be updated in tables
+        """
+        my_new_entity = {
+            u'PartitionKey': "{} {} {} Ownership".format(ownership["fullyqualifiednamespace"],
+        ownership["eventhubname"], ownership['consumergroup']),
+            u'RowKey': ownership['partitionid'],
+            u'ownerid' : ownership['ownerid'],
+        }
+        return my_new_entity
+
+    @classmethod
+    def _create_checkpoint_entity(cls, checkpoint):
+        """
+        create a dictionary with the new checkpoint attributes so that it can be updated in tables
+        """
+        my_new_entity = {
```

Comment:
Nit: Since you're creating a checkpoint entity, renaming this to `checkpoint_entity` is a bit more clear.

Reply:
yea I will replace checkpoint_entity that make sense